src/foam/dao/RestDAO.js

```diff
@@ -87,6 +88,14 @@ foam.CLASS({
        * Each key's value is network-foam-jsonified.
        */
       var query = [];
+
+      // Proxy non-serializable sinks. Otherwise, serialize and send the sink,
+      // then instantiate the results when they come back from the server.
+      var proxySink = this.Serializable.isInstance(sink) ? null :
```

Comment:
That would work for `networkSink`, but this is the local proxy. We want it to be:
(1) `null` iff `sink` is serializable (i.e., no local proxy; just use what comes back); otherwise:
(2) `sink` if `sink` is defined; otherwise:
(3) an `ArraySink`.

Further down, we only send `sink` when `proxySink` is **not** defined; i.e., `sink` is serializable.